torchgeo/datasets/idtrees.py

```diff
@@ -0,0 +1,553 @@
+        chm = self._load_image(path.replace("RGB", "CHM"))
+        las = self._load_las(path.replace("RGB", "LAS").replace(".tif", ".las"))
+        sample = {"image": image, "hsi": hsi, "chm": chm, "las": las}
+
+        if self.split == "test":
+            if self.task == "task2":
+                sample["boxes"] = self._load_boxes(path)
+        else:
+            sample["boxes"] = self._load_boxes(path)
+            sample["label"] = self._load_target(path)
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.images)
+
+    def _load_image(self, path: str) -> Tensor:
+        """Load a tiff file.
+
+        Args:
+            path: path to .tif file
+
+        Returns:
+            the image
+        """
+        with rasterio.open(path) as f:
+            array = f.read(out_shape=self.image_size, resampling=Resampling.bilinear)
+        tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+        return tensor
+
+    def _load_las(self, path: str) -> Tensor:
+        """Load a single point cloud.
+
+        Args:
+            path: path to .las file
+
+        Returns:
+            the point cloud
+        """
+        import laspy
+
+        las = laspy.read(path)
+        array = np.stack([las.x, las.y, las.z], axis=0)
+        tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+        return tensor
+
+    def _load_boxes(self, path: str) -> Tensor:
+        """Load object bounding boxes.
+
+        Args:
+            path: path to .tif file
+
+        Returns:
+            the bounding boxes
+        """
+        base_path = os.path.basename(path)
+
+        # Find object ids and geometries
+        if self.split == "train":
+            indices = self.labels["rsFile"] == base_path
+            ids = self.labels[indices]["id"].tolist()
+            geoms = [self.geometries[i]["geometry"]["coordinates"][0][:4] for i in ids]
+        # Test set - Task 2 has no mapping csv. Mapping is inside of geometry
+        else:
+            ids = [
+                k
+                for k, v in self.geometries.items()
+                if v["properties"]["plotID"] == base_path
+            ]
+            geoms = [self.geometries[i]["geometry"]["coordinates"][0][:4] for i in ids]
+
+        # Convert to pixel coords
+        boxes = []
+        with rasterio.open(path) as f:
+            for geom in geoms:
+                coords = [f.index(x, y) for x, y in geom]
+                xmin = min([coord[0] for coord in coords])
+                xmax = max([coord[0] for coord in coords])
+                ymin = min([coord[1] for coord in coords])
+                ymax = max([coord[1] for coord in coords])
+                boxes.append([xmin, ymin, xmax, ymax])
```

Comment:
@isaaccorley Had a question about the bounding box representation here. rasterio's index method is used to convert from (lon,lat) to pixel coordinates. However I noticed that this occasionally returns negative bbox coordinates. 

torchvision's `draw_bounding_boxes` states that it accepts box coordinates to be w.r.t image i.e. they must satisfy `0 <= xmin < xmax < W` and `0 <= ymin < ymax < H`. Is this not an issue?

I looked into this because I recently tried training an object detector (torchvision's Faster RCNN) on this dataset and the result looked a bit weird. In the docs they mention they need bboxes to satisfy the above constraint so I was wondering if this was the cause of my problem.

Reply:
Think there might be an issue.

The following is a comparison of the image  - **MLBS_5**

Left is QGIS (polygons from train_MLBS), right is plot by torchgeo

![image](https://user-images.githubusercontent.com/32295036/177311398-e2255ff5-c5de-4ecb-bc6a-6add05ba416d.png)